modules/core/src/main/java/org/apache/synapse/message/processor/impl/forwarder/ForwardingService.java

```diff
@@ -94,54 +110,76 @@
      */
     private long throttlingInterval = -1;
 
-    /** Configuration to continue the message processor even without stopping
-     * the message processor after maximum number of delivery */
+    // Message Queue polling interval value.
+    private long interval;
+
+    /*
+     * Configuration to continue the message processor even without stopping
+     * the message processor after maximum number of retry attempts.
+     */
     private boolean isMaxDeliveryAttemptDropEnabled = false;
 
-    public void execute(JobExecutionContext jobExecutionContext) throws JobExecutionException {
+    private SynapseEnvironment synapseEnvironment;
 
-        init(jobExecutionContext);
+    private boolean initialized = false;
 
+    public ForwardingService(MessageProcessor messageProcessor, BlockingMsgSender sender,
+                             SynapseEnvironment synapseEnvironment, long threshouldInterval) {
+        this.messageProcessor = messageProcessor;
+        this.sender = sender;
+        this.synapseEnvironment = synapseEnvironment;
+        // Initializes the interval to the Threshould interval value.
+        this.interval = threshouldInterval;
+    }
+
+    /**
+     * Starts the execution of this task which grabs a message from the message
+     * queue and dispatch it to a given endpoint.
+     */
+    public void execute() {
+        final long startTime = new Date().getTime();
```

Comment:
This issue was fixed in pull request https://github.com/wso2/wso2-synapse/pull/242